cmd/log.go

```diff
@@ -0,0 +1,58 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/redhat-developer/odo/pkg/application"
+	"github.com/redhat-developer/odo/pkg/component"
+	"github.com/redhat-developer/odo/pkg/logs"
+	"github.com/redhat-developer/odo/pkg/project"
+	"github.com/spf13/cobra"
+)
+
+var logCmd = &cobra.Command{
+	Use:   "log [component_name]",
+	Short: "Retrieve the log for the given component.",
+	Long:  `Retrieve the log for the given component.`,
+	Example: `  # Get the logs for the nodejs component,
+  odo log nodejs
+	`,
+	Args: cobra.RangeArgs(0, 1),
+	Run: func(cmd *cobra.Command, args []string) {
+
+		// Retrieve the client
+		client := getOcClient()
+
+		// Application
+		currentApplication, err := application.GetCurrent(client)
+		checkError(err, "")
+
+		// Project
+		currentProject := project.GetCurrent(client)
+
+		// Retrieve and set the current component if none passed
+		var currentComponent string
+		if len(args) == 0 {
```

Comment:
yes, but if `inputComponent` is empty than it returns the current component, otherwise it checks if inputComponent exists.

This is exactly what you are doing here with `args[0]` no?

wouldn't this have the same result?
```go
currentComponent := ""
if len(args) == 1 {
  currentComponent = args[1]
}
currentComponent = getComponent(client, currentComponent, currentApplication, currentProject)

 

```

Reply:
@kadel We're passing in blank for currentComponent.. which feels as if it's a undocumented function since it will return the active component rather than "getting" the component that's being passed in (currentComponent). Does that make sense?

Reply:
It is documented in a comment for that function.

I agree that it feels weird, but we have this logic in a lot of places, and it actually makes sense to have this abstracted away in one common function. Maybe a different name or small refactoring of that function might help.